packages/constants/src/simpleBlocks/astToSimpleBlocks.ts

```diff
@@ -0,0 +1,318 @@
+                  errors,
+                }),
+              )
+            } else if (isContentBlock(childTag)) {
+              stepChildren.push(
+                createContentBlock({
+                  prompt,
+                  tag: childTag,
+                  generateIdFn,
+                  errors,
+                }),
+              )
+            } else {
+              const content = convertElementToText(childTag, true)
+              if (content) {
+                stepChildren.push({
+                  id: generateIdFn(),
+                  type: 'text',
+                  content: content,
+                })
+              }
+            }
+          } else {
+            const content = nodeToText(child, true)
+            if (content) {
+              stepChildren.push({
+                id: generateIdFn(),
+                type: 'text',
+                content: content,
+              })
+            }
+          }
+        }
+
+        const stepAttributes = getStepAttributes({ tag, prompt })
+        blocks.push({
+          id: generateIdFn(),
+          type: 'step',
+          children: stepChildren,
+          attributes: stepAttributes,
+          ...(blockErrors.length > 0 && { errors: blockErrors }),
+        })
+      } else {
+        const childBlocks = processNodes({
+          nodes: tag.children,
+          generateIdFn,
+          prompt,
+          errors,
+        })
+        blocks.push(...childBlocks)
+      }
+      i++
+    } else {
+      // Accumulate consecutive non-ElementTag nodes into a single block
+      const accumulatedContent: string[] = []
+
+      while (i < nodes.length) {
+        const currentNode = nodes[i]!
+
+        // Stop if we hit an ElementTag that creates a block
+        if (currentNode.type === 'ElementTag') {
+          const tag = currentNode as ElementTag
+          if (isTopLevelBlock(tag)) {
+            break
+          }
+        }
+
+        // Accumulate content from this node
+        if (currentNode.type === 'Config') {
+          // Skip config nodes in content accumulation
+        } else {
+          accumulatedContent.push(nodeToText(currentNode))
+        }
+
+        i++
+      }
+
+      const content = accumulatedContent.join('')
+
+      if (content) {
+        const systemTagMatch = content.match(/^<s>([\s\S]*?)<\/system>/)
+        if (systemTagMatch) {
+          blocks.push({
+            id: generateIdFn(),
+            type: 'system',
+            children: [],
+          })
+        } else {
+          blocks.push({
+            id: generateIdFn(),
+            type: 'text',
+            content: content,
+          })
+        }
+      }
+    }
+  }
+
+  return blocks
+}
+
+export function astToSimpleBlocks({
+  ast,
+  prompt,
+  errors = [],
+}: {
+  ast: Fragment
+  prompt: string
+  errors?: AstError[]
+}): AnyBlock[] {
+  const generateIdFn = createIdGenerator()
+  return processNodes({ nodes: ast.children, generateIdFn, prompt, errors })
```

Comment:
no point of processNodes function since it's a local function and you are immediately calling it